src/libraries/Common/src/System/SR.cs

```diff
@@ -10,14 +10,30 @@ namespace System
 {
     internal partial class SR
     {
+#if (!NETSTANDARD1_0 && !NETSTANDARD1_1 && !NET45) // AppContext is not supported on < NetStandard1.3 or < .NET Framework 4.5 so no need to cache the value.
+        // cache result of AppContext Switch System.Resources.UsingResourceKeys
+        private static bool? s_usingResourceKeys;
+#endif
+
         // This method is used to decide if we need to append the exception message parameters to the message when calling SR.Format.
         // by default it returns false.
         // Native code generators can replace the value this returns based on user input at the time of native code generation.
         // Marked as NoInlining because if this is used in an AoT compiled app that is not compiled into a single file, the user
         // could compile each module with a different setting for this. We want to make sure there's a consistent behavior
         // that doesn't depend on which native module this method got inlined into.
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static bool UsingResourceKeys() => false;
+        private static bool UsingResourceKeys()
+        {
+#if (!NETSTANDARD1_0 && !NETSTANDARD1_1 && !NET45) // AppContext is not supported on < NetStandard1.3 or < .NET Framework 4.5
+            if (s_usingResourceKeys == null)
```

Comment:
Nullable is not atomic. This has thread safety bug. You should use the same pattern as what we use in other places with AppContext.

Reply:
I am not sure whether you need to manually cache this lazily here at all. Can you just depend on lazy execution of the static constructor to cache the value?

Reply:
Looks like other places where we use AppContext in corelib(GlobalizationMode, SerializationInfo, AppDomain, etc.) we don't cache values as looks like AppContext is already doing that so I can just remove the caching aspect here. FWIW we are also not caching in PR https://github.com/dotnet/runtime/pull/38129 where we are adding another feature switch which also has runtimeconfig capabilities.

Reply:
To achieve the caching don't you need to include a file like https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/LocalAppContextSwitches.cs (it would be in every assembly I guess)

Reply:
> Looks like other places where we use AppContext in corelib(GlobalizationMode, SerializationInfo, AppDomain, etc.) we don't cache

I am not sure what you have looked at. Both GlobalizationMode and SerializationInfo are caching the lookup...

Reply:
> it would be in every assembly I guess

I do not think we would want to add an extra type to every assembly just to cache this.

Reply:
Oh I was only looking at GlobalizationMode.GetSwitchValue implementation, but now I see that this gets called by GlobalizationMode static constructor, which initializes the value and caches it in the Invariant property.

Reply:
> I do not think we would want to add an extra type to every assembly just to cache this.
 
yes the LocalAppContextSwitches mechanism seems a bit over engineered to me.